ansible-ipi-install/roles/installer/tasks/25c_disconnected_registry_create.yml

```
@@ -0,0 +1,186 @@
+---
+
+- name: Create a disconnected registry
+  block:
+    - name: Make sure needed packages are installed
+      package:
+        state: present
+        name: "{{ item }}"
+      loop:
+        - "{{ packages_registry | default([]) }}"
+      become: true
+    
+    - name: open firewall port {{ registry_port }}
+      firewalld:
+        port: "{{ registry_port }}/tcp"
+        permanent: yes
+        immediate: yes
+        state: enabled
+      become: true
+    
+    - name: query getent for homedir
+      getent:
+        database: passwd
+        key: "{{ ansible_ssh_user }}"
+    
+    - name: set homedir fact
+      set_fact:
+        homedir: "{{ getent_passwd | json_query('*[4]') | first }}"
+    
+    - name: Create directory to hold the registry files
+      file:
+        path: "{{ item }}"
+        state: directory
+        owner: "{{ ansible_user }}"
+        group: "{{ ansible_user }}"
+        mode: u=rwX,g=rX,o=rX
+        recurse: yes
+      loop:
+        - "{{ registry_dir_auth }}"
+        - "{{ registry_dir_cert }}"
+        - "{{ registry_dir_data }}"
+      become: true
+    
+    - name: Generate htpasswd entry
+      command: htpasswd -bBn dummy dummy
+      register: htpass_entry
+    
+    - name: Write htpasswd file
+      copy:
+        content: '{{ htpass_entry.stdout }}'
+        dest: "{{ registry_dir_auth }}/htpasswd"
+        owner: "{{ ansible_user }}"
+        group: "{{ ansible_user }}"
+    
+    - name: Create registry auth for pullsecret
+      set_fact:
+        disconnected_auth: '  {"{{ ansible_fqdn }}:{{ registry_port }}": {"auth": "ZHVtbXk6ZHVtbXk=", "email": "{{ ansible_user }}@redhat.com" } }'
```

Would be great to have ability to specify alternate name for registry, and if not set fallback to `ansible_fqdn`
```suggestion
        disconnected_auth: '  {"{{ registry_name|default(ansible_fqdn) }}:{{ registry_port }}": {"auth": "ZHVtbXk6ZHVtbXk=", "email": "{{ ansible_user }}@redhat.com" } }'
```

---

i was looking at this and wondering why would you need to specify a alternate name when creating the disconnected registry? The disconnected registry should have a FQDN here for the server the registry is being created on. ansible_fqdn provides this for the particular server.

I believe this can be done by changing a couple of variables, but was wondering if it was needed.